tfhe/src/integer/server_key/radix_parallel/add.rs

```diff
@@ -1,45 +1,45 @@
 use crate::core_crypto::commons::numeric::UnsignedInteger;
 use crate::integer::ciphertext::IntegerRadixCiphertext;
-use crate::integer::server_key::radix_parallel::sub::SignedOperation;
 use crate::integer::{BooleanBlock, RadixCiphertext, ServerKey, SignedRadixCiphertext};
 use crate::shortint::ciphertext::Degree;
 use crate::shortint::Ciphertext;
 use rayon::prelude::*;
 
-#[repr(u64)]
-#[derive(PartialEq, Eq)]
-pub(crate) enum OutputCarry {
-    /// The block does not generate nor propagate a carry
-    None = 0,
-    /// The block generates a carry
-    Generated = 1,
-    /// The block will propagate a carry if it ever
-    /// receives one
-    Propagated = 2,
+/// Possible output flag that the advanced_add_assign_with_carry family of
+/// functions can compute.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub(crate) enum OutputFlag {
+    /// Request no flag at all
+    None,
+    /// The overflow flag is the flag that tells whether the input carry bit onto the last bit
+    /// is different than the output bit.
+    ///
+    /// This is usefull to know if a signed addition overflowed (in 2's complement)
+    Overflow,
+    /// The carry flag is simply the carry bit that the output from the last pair of blocks
+    /// in an addtion.
+    ///
+    /// This is useful to know if an unsigned addition overflowed.
+    Carry,
 }
 
-/// Function to create the LUT used in parallel prefix sum
-/// to compute carry propagation
-///
-/// If msb propagates it take the value of lsb,
-/// this means:
-/// - if lsb propagates, msb will propagate (but we don't know yet if there will actually be a carry
-///   to propagate),
-/// - if lsb generates a carry, as msb propagates it, lsb will generate a carry. Note that this lsb
-///   generates might be due to x propagating ('resolved' by an earlier iteration of the loop)
-/// - if lsb does not output a carry, msb will have nothing to propagate
-///
-/// Otherwise, msb either does not generate, or it does generate,
-/// but it means it won't propagate
-fn prefix_sum_carry_propagation(msb: u64, lsb: u64) -> u64 {
-    if msb == OutputCarry::Propagated as u64 {
-        lsb
-    } else {
-        msb
+impl OutputFlag {
+    /// Returns which flag shall be computed in order to get the flag
+    /// telling the overflow status
+    pub(crate) const fn from_signedness(is_signed: bool) -> Self {
+        if is_signed {
+            Self::Overflow
+        } else {
+            Self::Carry
+        }
     }
 }
 
-fn should_hillis_steele_propagation_be_faster(num_blocks: usize, num_threads: usize) -> bool {
+fn should_parallel_propagation_be_faster(
```

Comment:
This is still use in the function that dispatches between using the regular sequential propagation algo or the parallel one, depending on the number of threads and blocks

So that consumer grade laptop will generally run the sequential algorithm because the parallel one requires more threads than generally available